nisystemlink/clients/dataframe/_data_frame_client.py

```diff
@@ -13,11 +13,16 @@
 )
 from nisystemlink.clients.core.helpers import IteratorFileLike
 from requests.models import Response
-from uplink import Body, Field, Path, Query
+from uplink import Body, Field, Path, Query, retry
 
 from . import models
 
 
+@retry(
+    when=retry.when.status([429, 500, 502, 503, 504]),
```

Comment:
Don't retry 500 responses, its not clear that is safe to do.
```suggestion
    when=retry.when.status([429, 502, 503, 504]),
```